x/axelarnet/message_handler.go

```diff
@@ -0,0 +1,285 @@
+package axelarnet
+
+import (
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ibctransfertypes "github.com/cosmos/ibc-go/v4/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v4/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v4/modules/core/exported"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/axelarnetwork/axelar-core/utils/events"
+	"github.com/axelarnetwork/axelar-core/x/axelarnet/keeper"
+	"github.com/axelarnetwork/axelar-core/x/axelarnet/types"
+	nexus "github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+// Message is attached in ICS20 packet memo field
+type Message struct {
+	DestinationChain   string `json:"destination_chain"`
+	DestinationAddress string `json:"destination_address"`
+	Payload            []byte `json:"payload"`
+	Type               int    `json:"type"`
```

Comment:
```suggestion
	Type               int64    `json:"type"`
```
`int64` or `int32` perhaps so it's not platform dependent when decoding?